app/controllers/pieces_controller.rb

```diff
@@ -12,10 +12,16 @@ def update
       flash[:danger] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was invalid"
     end
 
+    if @piece.game.en_passant_opportunity_active?
+      @piece.game.update_attribute(:allows_en_passant_capture?, true)
+    else
+      @piece.game.update_attribute(:allows_en_passant_capture?, false) if @piece.game.allows_en_passant_capture?
+    end
+
```

Comment:
Your question is a bit vague. What exactly are you suggesting? The only comment I can offer right now is all the code in pieces#update is there to represent the various things that can/should happen within the context of a player turn.

I'll explain/elaborate further once I have a better understanding of what you're asking about.